### ISSUE TITLE
HI, How to use these API?

### ISSUE THREAD

**Issue by ckvir** [erkia/cli#1]
HI, Could you tell me how to use these API function?
Do you have some sample code to demo?

Thanks

**Comment by erkia** [comment]
Hi,
this CLI interface is mainly meant for embedded systems, but I've added a demo for Linux: [https://github.com/erkia/demos/tree/master/cli](https://github.com/erkia/demos/tree/master/cli)

**Comment by ckvir** [comment]
Hi, 
Thanks! That's worked. 
But how to use AutoComplete & CmdHistory?
Many thanks

**Comment by erkia** [comment]
Autocomplete works by pressing TAB (twice if there is ambiguity) and history is navigated by UP and DOWN keys.

**Comment by ckvir** [comment]
Hi I found that i cannot use left or right arrow to move cursor?

**Comment by erkia** [comment]
Yes, moving cursor and inserting/deleting characters other than at the end is not supported.
